ibis/expr/rules.py

```diff
@@ -147,7 +147,7 @@ def member_of(obj, arg):
 
 
 @validator
-def list_of(inner, arg, min_length=0):
+def list_of(inner, arg, min_length=0, coerce_to=ir.sequence):
```

Comment:
This validator actually returns with an `ir.ValueList` so the naming could be better, like `list_of_exprs`. We may revisit this later.

Reply:
I suppose one could implement `value_list` etc in terms of `list_of`, now that you've added a `coerce_to` argument. OTOH removing some of the `*List*` APIs would probably be better than having a few more specific validators.

Reply:
I'm not really satisfied with this keyword argument, so I'm open to alternatives. Perhaps https://github.com/ibis-project/ibis/issues/3000 will clean this up a little bit.